documentation/API.md

```diff
@@ -352,6 +383,88 @@ For displaying custom data (e.g. Users, Products, Orgs). User's JWT determines w
+
+
+##### RESPONSE Body (example):
+
+```JSON
+{
+    "displayData": [
+        {
+            "type": "NOTIFICATION",
+            "status": "SUCCESS",
+            "displayString": "Congratulations, application S-UUR-0001 has been approved",
+            "text": "## Your product registration license ...",
+            "errorLog": null
+        }
+    ],
+    "actionsOutput": [
+        {
+            "action": "sendNotification",
+            "status": "SUCCESS",
+            "output": {
+                "notification": {
+                    "id": 2,
+                    "user_id": 9,
+                    "application_id": 239,
+                    "review_id": 3,
+                    "email_recipients": "carl@msupply.foundation",
+                    "subject": "Congratulations, application S-UUR-0001 has been approved",
+                    "message": "## Your product registration license ...",
+                    "attachments": [],
+                    "email_sent": false,
+                    "is_read": false
+                }
+            },
+            "errorLog": null
+        }
+    ]
+}
+```
+
+`displayData` is used by the front-end to present the Preview results in the UI. `actionsOutput` contains the raw output of each action that ran.
+
+
+
+### Extend application deadline
+
+POST: `/extend-application`
+
+Endpoint for extending a deadline associated with an application. Currently, the main use case is for extending an applicant deadline for responding to a request for changes (event code: `applicantDeadline`), but in theory could be used for extending other types of deadlines as well.
+
+It works by finding an event in the `trigger_schedule` table with matching `applicationId` and `eventCode` and then extending the `time_scheduled` by the specified amount. It also simulates a trigger `ON_EXTEND` which can be used to trigger other events in the application (such as resetting the outcome from "EXPIRED" back to "PENING" for example).
+
+##### REQUEST parameters:
+
+- `applicationId`
+- `eventCode` -- value in `trigger_schedule` to match. For applicant deadlines we are currently using the hard-coded value `applicantDeadline` to match in the front-end UI. This event code is also passed along with the "ON_EXTEND" trigger to use to match to actions. 
+- `extensionTime` -- a length of time: either a number (which will be interpreted as days), or a [Luxon duration object](https://moment.github.io/luxon/api-docs/index.html#duration). If the current deadline has already expired, this extension time will be added to the current time. If it's not already expired, it'll be added to the current `time_scheduled`.
```

Comment:
```suggestion
- `extensionTime` -- a length of time: either a number (which will be interpreted as days), or a [Luxon duration object](https://moment.github.io/luxon/api-docs/index.html#duration). If the current deadline has already expired, this extension time will be added to the current time. If it's not already expired, it'll be added to the `time_scheduled`.
```

Reply:
Okay, I think this is clearer now: cf87a08e